pkg/protoassert/assert.go

```diff
@@ -0,0 +1,82 @@
+package protoassert
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+)
+
+func Equal(t *testing.T, expected, actual proto.Message, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	e, err := toJson(expected)
+	require.NoError(t, err)
+	a, err := toJson(actual)
+	require.NoError(t, err)
+	return assert.JSONEq(t, e, a, msgAndArgs)
+}
+
+func SlicesEqual[T proto.Message](t *testing.T, expected, actual []T, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	areEqual := assert.Len(t, actual, len(expected), msgAndArgs)
+	for i, e := range expected {
+		a := actual[i]
+		areEqual = Equal(t, a, e, msgAndArgs) && areEqual
+	}
+	if !areEqual {
+		t.Log(msgAndArgs...)
+	}
+	return areEqual
+}
+
+func MapSliceEqual[K comparable, T proto.Message](t *testing.T, expected, actual map[K][]T, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	ek := maps.Keys(expected)
+	ea := maps.Keys(actual)
```

Comment:
nit: rather `ak` for actual keys (no actual need for change, though). Would also apply to `MapEqual`.